### PR TITLE
Tweak: Eicons to SVG [ED-2650]

### PR DESCRIPTION
**Pull request by rotemee**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: core/page-assets/data-managers/font-icon-svg/font-awesome.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Font Awesome Icon Svg.
+ *
+ * @since 3.4.0
+ */
+class Font_Awesome extends Base {
+	const LIBRARY_CURRENT_VERSION = 5;
```
Comment (udidol) [inline core/page-assets/data-managers/font-icon-svg/font-awesome.php:14]: Same here, full version number as string - `'5.15.3'`

**Review by udidol — CHANGES_REQUESTED** [review]

File: assets/dev/js/frontend/utils/lightbox/lightbox.js [hunk 2]
```diff
@@ -1,4 +1,19 @@
...
+	facebook,
+	frameExpand,
+	frameMinimize,
+	loading,
+	pinterest,
+	shareArrow,
+	twitter,
+	zoomInBold,
+	zoomOutBold,
+} from '@elementor/e-icons';
```
[the remark this answers is not in the document]
Comment (rotemee) [reply assets/dev/js/frontend/utils/lightbox/lightbox.js:16]: @udidol If I will add them all at once (*), and someone will add more icons to the existing icons list, they will also be imported to the lightbox, although only these specific icons are needed.

File: assets/dev/js/frontend/utils/icons/manager.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
+export default class IconsManager {
+	static symbolsContainer;
+
+	static iconsUsageList = [];
```
[the remark this answers is not in the document]
Comment (rotemee) [reply assets/dev/js/frontend/utils/icons/manager.js:4]: Fixed.

File: assets/dev/js/frontend/utils/lightbox/lightbox.js [hunk 4]
```diff
@@ -78,11 +93,20 @@ module.exports = elementorModules.ViewModule.extend( {
 	},
 
 	initModal: function() {
+		const closeIcon = {};
+
+		// If the experiment is active the closeIcon should be an entire SVG element otherwise it should pass the eicon class name.
+		if ( elementorFrontend.config.experimentalFeatures.e_font_icon_svg ) {
+			closeIcon.iconElement = close.element;
```
[the remark this answers is not in the document]
Comment (rotemee) [reply assets/dev/js/frontend/utils/lightbox/lightbox.js:100]: @udidol The `close` is the icon element that was imported to this file.

File: assets/dev/js/frontend/utils/lightbox/lightbox.js [hunk 5]
```diff
@@ -352,30 +393,76 @@ module.exports = elementorModules.ViewModule.extend( {
...
+				},
+				expandAttrs = {
+					...fullScreenAttrs,
+				};
+
+			if ( ! elementorFrontend.config.experimentalFeatures.e_font_icon_svg ) {
+				expandAttrs.class = slideshowClasses.iconExpand;
+			}
+
+			elements.$iconExpand = $( iconElement ).append( $( '<span>' ), $( '<span>' ) ).attr( expandAttrs ).on( 'click', this.toggleFullscreen );
```
[the remark this answers is not in the document]
Comment (rotemee) [reply assets/dev/js/frontend/utils/lightbox/lightbox.js:453]: Fixed.

File: core/page-assets/data-managers/font-icon-svg/font-awesome.php (hunk 1 above)
Comment (rotemee) [reply core/page-assets/data-managers/font-icon-svg/font-awesome.php:14]: Fixed.

**Review by udidol — APPROVED** [review]

File: assets/dev/js/frontend/utils/icons/manager.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+			IconsManager.symbolsContainer = document.getElementById( symbolsContainerId );
+
+			if ( ! IconsManager.symbolsContainer ) {
+				IconsManager.symbolsContainer = document.createElementNS( 'http://www.w3.org/2000/svg', 'svg' );
+
+				IconsManager.symbolsContainer.setAttributeNS( null, 'style', 'display: none;' );
+				IconsManager.symbolsContainer.setAttributeNS( null, 'class', symbolsContainerId );
+
+				document.body.appendChild( IconsManager.symbolsContainer );
```
Comment (kobizz) [inline assets/dev/js/frontend/utils/icons/manager.js:20]: Create a dedicated method

File: assets/dev/js/frontend/utils/icons/manager.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+		if ( ! IconsManager.iconsUsageList.includes( elementName ) ) {
+			if ( ! IconsManager.symbolsContainer.querySelector( elementSelector ) ) {
+				const symbol = document.createElementNS( 'http://www.w3.org/2000/svg', 'symbol' );
+
+				symbol.id = elementName;
+				symbol.innerHTML = '<path d="' + path + '"></path>';
+
+				symbol.setAttributeNS( null, 'viewBox', '0 0 ' + width + ' ' + height );
+
+				IconsManager.symbolsContainer.appendChild( symbol );
```
Comment (kobizz) [inline assets/dev/js/frontend/utils/icons/manager.js:39]: Create a dedicated method

File: assets/dev/js/frontend/utils/lightbox/lightbox.js [hunk 8]
```diff
@@ -451,18 +542,34 @@ module.exports = elementorModules.ViewModule.extend( {
 	activateFullscreen: function() {
 		const classes = this.getSettings( 'classes' );
 		screenfull.request( this.elements.$container.parents( '.dialog-widget' )[ 0 ] );
-		this.elements.$iconExpand.removeClass( classes.slideshow.iconExpand )
-			.addClass( classes.slideshow.iconShrink )
-			.attr( 'aria-checked', 'true' );
+
+		if ( elementorFrontend.config.experimentalFeatures.e_font_icon_svg ) {
+			this.elements.$iconExpand.addClass( classes.hidden ).attr( 'aria-checked', 'true' );
+			this.elements.$iconMinimize.removeClass( classes.hidden ).attr( 'aria-checked', 'true' );
```
Comment (kobizz) [inline assets/dev/js/frontend/utils/lightbox/lightbox.js:548]: Check why do you need to add 'aria-checked' for both elements

File: assets/dev/js/frontend/utils/lightbox/lightbox.js [hunk 9]
```diff
@@ -451,18 +542,34 @@ module.exports = elementorModules.ViewModule.extend( {
...
 	deactivateFullscreen: function() {
 		const classes = this.getSettings( 'classes' );
 		screenfull.exit();
-		this.elements.$iconExpand.removeClass( classes.slideshow.iconShrink )
-			.addClass( classes.slideshow.iconExpand )
-			.attr( 'aria-checked', 'false' );
+
+		if ( elementorFrontend.config.experimentalFeatures.e_font_icon_svg ) {
+			this.elements.$iconExpand.removeClass( classes.hidden ).attr( 'aria-checked', 'false' );
+			this.elements.$iconMinimize.addClass( classes.hidden ).attr( 'aria-checked', 'false' );
```
Comment (kobizz) [inline assets/dev/js/frontend/utils/lightbox/lightbox.js:565]: Same here

File: assets/dev/js/frontend/utils/lightbox/lightbox.js [hunk 10]
```diff
@@ -264,9 +288,18 @@ module.exports = elementorModules.ViewModule.extend( {
...
+					iconElement: facebook,
+				},
+				twitter: {
+					label: i18n.shareOnTwitter,
+					iconElement: twitter,
+				},
+				pinterest: {
+					label: i18n.pinIt,
+					iconElement: pinterest,
+				},
```
Comment (kobizz) [inline assets/dev/js/frontend/utils/lightbox/lightbox.js:302]: Where do you use the `label`?

File: assets/dev/js/frontend/utils/lightbox/lightbox.js [hunk 11]
```diff
@@ -358,30 +399,80 @@ module.exports = elementorModules.ViewModule.extend( {
...
+					role: 'switch',
+					'aria-checked': false,
+					'aria-label': i18n.fullscreen,
+				},
+				expandAttrs = {
+					...fullScreenAttrs,
+				};
+
+			// Only if the experiment is not active, we use the class-name in order to render the icon.
+			if ( ! elementorFrontend.config.experimentalFeatures.e_font_icon_svg ) {
```
Comment (kobizz) [inline assets/dev/js/frontend/utils/lightbox/lightbox.js:450]: Add a class property for `e_font_icon_svg`

**Review by kobizz — CHANGES_REQUESTED** [review]

File: assets/dev/js/frontend/utils/icons/manager.js (hunk 6 above)
Comment (rotemee) [reply assets/dev/js/frontend/utils/icons/manager.js:20]: Done.

File: assets/dev/js/frontend/utils/lightbox/lightbox.js (hunk 8 above)
Comment (rotemee) [reply assets/dev/js/frontend/utils/lightbox/lightbox.js:548]: Fixed (also added to the zoom buttons that were missing in some cases).

File: assets/dev/js/frontend/utils/lightbox/lightbox.js (hunk 11 above)
Comment (rotemee) [reply assets/dev/js/frontend/utils/lightbox/lightbox.js:450]: Done.

**Comment by EXECdesign** [comment]
Still the same issues in EP 3.5.1: eicons.woff2?5.10.0 gets loaded, Google Lighthouse test fails.

**Pull request merged**